### PR TITLE
feat: add ldproxy and cargo-espflash

### PR DESCRIPTION
**Pull request by pulsastrix**

### CODE REVIEW

File: overlay.nix [hunk 1]
```diff
@@ -42,6 +42,8 @@ rec {
   # LLVM
   llvm-xtensa = prev.callPackage ./pkgs/llvm-xtensa-bin.nix { };
   llvm-xtensa-lib = prev.callPackage ./pkgs/llvm-xtensa-lib.nix { };
+  rust-ldproxy = prev.callPackage ./pkgs/rust-ldproxy.nix { };
+  rust-cargo-espflash = prev.callPackage ./pkgs/rust-cargo-espflash.nix { };
```
Comment (JKRhb) [inline overlay.nix]: Maybe these could be grouped under a different comment (as they are not LLVM-related)?

**Review by JKRhb — APPROVED** [review]
LGTM :) See one minor comment below.

**Comment by Kernald** [comment]
Is there any particular reason these are built manually rather than using the versions from nixpkgs?

For `ldproxy`, nixpkgs has had it for four months, and a more recent version: https://github.com/NixOS/nixpkgs/blob/d828bdc634fe4cb80cbce4d13287f6ea61fe9396/pkgs/by-name/ld/ldproxy/package.nix
And `cargo-espflash` has been renamed to `espflash` a while ago (https://github.com/NixOS/nixpkgs/pull/286386), with the same version that has been added in this PR available in nixpkgs months ago as well.

I'm definitely not trying to criticise this PR, I appreciate your work, but I'm trying to get this to work through nix-darwin, and those two packages work just fine on Darwin as packaged in nixpkgs, so I'm trying to understand the rationale behind having them re-defined in this repo.

**Comment by pulsastrix** [comment]
@Kernald TBH I am not sure why I did not consider using the packages in nixpkgs, it's possible I just wasn't aware that they existed back then.

I will test whether using the packages from `nixpkgs` causes any issues in my environment as soon as I have the time.
If there are no issues, there shouldn't be any reason not to use them.

**Comment by Kernald** [comment]
Thanks for the feedback! For what it's worth, I've been using the ones from nixpkgs (from `nixos-unstable` but I don't think they've changed at all since the NixOS 24.05 release) without any difficulties with an ESP32-S3. Happy to send a PR to use those and clean-up the package definitions here, and hopefully send some PRs soon to help with Darwin compatibility overall.

**Pull request merged**